### PR TITLE
refactor: 💡 Migrate SQFormMultiValue to TS

### PR DESCRIPTION
**Pull request by 20BBrown14**
✅ Closes: #409

### CODE REVIEW

File: src/components/SQForm/SQFormMultiValue.tsx [hunk 1]
```diff
@@ -250,76 +288,61 @@ function SQFormMultiValue({
...
+          const endAdornment = Object.assign<
+            Record<string, Record<string, unknown>>,
+            Record<string, Record<string, unknown>>
+          >({}, params.InputProps.endAdornment || {});
+          if (endAdornment && endAdornment.props) {
+            endAdornment.props = {
+              ...(endAdornment.props || {}),
+              className: `${endAdornment.props.className} ${clearButtonClasses.endAdornment}`,
+            };
+          }
```
Comment (20BBrown14) [inline src/components/SQForm/SQFormMultiValue.tsx:317]: I'm not so sure about this, particularly the type for `Object.assign`.

File: src/components/SQForm/SQFormMultiValue.tsx [hunk 2]
```diff
@@ -1,28 +1,59 @@
...
+import {useForm} from './useForm';
+import {BaseFieldProps, Option, optionValue} from 'types';
+import {
+  OuterElementContextInterface,
+  OuterElementTypeProps,
+} from './SQFormAutocomplete';
+
+interface SQFormMultiValueProps extends BaseFieldProps {
+  /** options to select from */
+  children: Option[];
```
Comment (codyarose) [inline src/components/SQForm/SQFormMultiValue.tsx:18]: Based on the previous propTypes, everything outside of `BaseFieldProps` should be optional

File: src/components/SQForm/SQFormMultiValue.tsx [hunk 3]
```diff
@@ -250,76 +288,61 @@ function SQFormMultiValue({
...
-            inputProps={{
-              ...params.inputProps,
-              disabled: isDisabled
-            }}
-            FormHelperTextProps={{error: isFieldError}}
-            helperText={!isDisabled && HelperTextComponent}
-          />
-        )}
+        renderInput={(params) => {
+          console.log('params', params);
```
Comment (codyarose) [inline src/components/SQForm/SQFormMultiValue.tsx:306]: Some leftover console.log and comments here

File: src/components/SQForm/SQFormMultiValue.tsx (hunk 1 above)
Comment (codyarose) [reply src/components/SQForm/SQFormMultiValue.tsx:317]: Oof, yeah it's kind of hard for me to follow what's going on with that at the moment, but maybe the `InputAdornmentProps` type could help here?

**Review by codyarose — CHANGES_REQUESTED** [review]

File: src/components/SQForm/SQFormMultiValue.tsx (hunk 1 above)
Comment (20BBrown14) [reply src/components/SQForm/SQFormMultiValue.tsx]: So I was able to come up with a different solution that is far easier to read and gets the job done

File: src/components/SQForm/SQFormMultiValue.tsx (hunk 2 above)
Comment (20BBrown14) [reply src/components/SQForm/SQFormMultiValue.tsx:19]: Yep, everything except Children. It should actually be required even if the array is empty.

**Comment by 20BBrown14** [comment]
Sorry it took so long to update. This is ready for review again with all pending change requests completed.

**Comment by 20BBrown14** [comment]
🎉  All CI checks passing, and the changes I want are in the PR. Finally ready.

File: src/components/SQForm/SQFormMultiValue.tsx [hunk 4]
```diff
@@ -115,6 +147,7 @@ function SQFormMultiValue({
   React.useEffect(() => {
     // Don't run this effect every time displayOptions changes
     // This also prevents infinite rerenders
+    // Maybe use deep-equal? This won't work if key order changes
```
Comment (hvilander) [inline src/components/SQForm/SQFormMultiValue.tsx:150]: Was this comment left in intentionally? Seem like it might be a note for you.

**Review by hvilander — APPROVED** [review]

File: src/components/SQForm/SQFormMultiValue.tsx (hunk 4 above)
Comment (20BBrown14) [reply src/components/SQForm/SQFormMultiValue.tsx:150]: ya, as a note in case we run into issues later on related to this `useEffect`.

**Review by SeanGroff — APPROVED** [review]
Awesome!

**Pull request merged**